File: java_solutions/144.binary-tree-preorder-traversal.java
/*
 * @lc app=leetcode id=144 lang=java
 *
 * [144] Binary Tree Preorder Traversal
 */

class Solution {
    public List<Integer> preorderTraversal(TreeNode root) {
        return root == null ? Collections.emptyList() : Stream.concat(Stream.of(root.val), Stream.concat(preorderTraversal(root.left).stream(), preorderTraversal(root.right).stream())).toList();
    }
}

File: kotlin_solutions/144.binary-tree-preorder-traversal.kt
/*
 * @lc app=leetcode id=144 lang=kotlin
 *
 * [144] Binary Tree Preorder Traversal
 */

class Solution {fun preorderTraversal(root: TreeNode?): List<Int> =root?.let { listOf(it.`val`) + preorderTraversal(it.left) + preorderTraversal(it.right) } ?: emptyList()}
